src/commands/revisionDraft/RevisionDraftUpdateBaseStep.ts

```diff
@@ -34,7 +35,13 @@ export abstract class RevisionDraftUpdateBaseStep<T extends IContainerAppContext
     private initRevisionDraftTemplate(): Template {
         let template: Template | undefined = ext.revisionDraftFileSystem.parseRevisionDraft(this.baseItem);
         if (!template) {
-            template = nonNullValueAndProp(this.baseItem.revision, 'template');
+            // Branching path reasoning: https://github.com/microsoft/vscode-azurecontainerapps/blob/main/src/commands/revisionDraft/README.md
+            // Make deep copies so we don't accidentally modify the cached values
+            if (ContainerAppItem.isContainerAppItem(this.baseItem) || this.baseItem.containerApp.revisionsMode === KnownActiveRevisionsMode.Single) {
```

Comment:
Can you not use the `getParentResourceFromItem` util function here to pick between containerApp or revision?

Reply:
Oh yeah good catch, I think you can